src/server/engine_shard_set.cc

```diff
@@ -448,18 +448,17 @@ void EngineShard::PollExecution(const char* context, Transaction* trans) {
   ShardId sid = shard_id();
   stats_.poll_execution_total++;
 
-  // Check if the caller was handled by a previous poll.
-  if (trans && !trans->IsArmedInShard(sid))
-    return;
+  uint16_t local_mask = trans ? trans->DisarmInShardWhen(sid, Transaction::AWAKED_Q) : 0;
```

Comment:
`local` here does not provide much value, on the other hand maybe call it `trans_mask` instead?